addons/email_template/email_template.py

```diff
@@ -505,6 +505,11 @@ def generate_email_batch(self, cr, uid, template_id, res_ids, context=None, fiel
             # Add report in attachments: generate once for all template_res_ids
             if template.report_template:
                 for res_id in template_res_ids:
+                    #Fix generate report for relevant partner only
+                    try:
+                        del ctx['active_ids']
+                    except:
```

Comment:
One good practice when using exceptions is, don't catch an exception if you know exactly what the problem could be. In this case, the key could be missing from the dictionary. Check for that one condition instead and in that case, just don't try to remove it.